SPECS/td-agent/td-agent.patch

```diff
@@ -0,0 +1,263 @@
+diff --git a/td-agent/Rakefile b/td-agent/Rakefile
+index 24f56f5..a9765f9 100755
+--- a/td-agent/Rakefile
++++ b/td-agent/Rakefile
+@@ -146,8 +146,10 @@ class DownloadTask
+     url_base = "https://github.com/jemalloc/jemalloc/releases/download/"
+     @file_jemalloc_source = File.join(DOWNLOADS_DIR, filename)
+     file @file_jemalloc_source do
+-      url = "#{url_base}/#{version}/#{filename}"
+-      download_file(url, filename)
++      ensure_directory(DOWNLOADS_DIR) do
++        cp("/usr/src/mariner/SOURCES/#{filename}", DOWNLOADS_DIR)
++        sh("tar", "xvf", "#{filename}")
++      end
+     end
+   end
+ 
+@@ -167,7 +169,10 @@ class DownloadTask
+     @file_ruby_source = File.join(DOWNLOADS_DIR, filename)
+ 
+     file @file_ruby_source do
+-      download_file(url, filename, sha256sum)
++      ensure_directory(DOWNLOADS_DIR) do
++        cp("/usr/src/mariner/SOURCES/#{filename}", DOWNLOADS_DIR)
++        sh("tar", "xzvf", "#{filename}")
++      end
+     end
+   end
+ 
+@@ -181,7 +186,9 @@ class DownloadTask
+     @file_ruby_installer_x64 = File.join(DOWNLOADS_DIR, filename)
+ 
+     file @file_ruby_installer_x64 do
+-      download_file(url, filename, sha256sum)
++      ensure_directory(DOWNLOADS_DIR) do
++        cp("/usr/src/mariner/SOURCES/#{filename}", DOWNLOADS_DIR)
++      end
+     end
+   end
+ 
+@@ -192,12 +199,8 @@ class DownloadTask
+         dirname = "fluentd-#{FLUENTD_REVISION}"
+         rm_rf("fluentd") if File.exists?("fluentd")
+         rm_rf(dirname) if File.exists?(dirname)
+-        sh("git", "clone", "https://github.com/fluent/fluentd.git")
+-        cd("fluentd") do
+-          sh("git", "checkout", FLUENTD_REVISION)
+-        end
+-        mv("fluentd", dirname)
+-        sh("tar", "cvfz", "#{dirname}.tar.gz", dirname)
++        cp("/usr/src/mariner/SOURCES/#{dirname}.tar.gz", DOWNLOADS_DIR)
++        sh("tar", "xzvf", "#{dirname}.tar.gz")
+       end
+     end
+   end
+@@ -264,12 +267,66 @@ class BuildTask
+ 
+       desc "Install core_gems"
+       task :core_gems => [:"download:core_gems", :ruby] do
+-        install_gems(@download_task.files_core_gems)
++        cd(DOWNLOADS_DIR) do
++          ensure_directory("core_gems") do
++            cp("/usr/src/mariner/SOURCES/gemfile.tar.gz", ".")
++            sh("tar", "xvf", "gemfile.tar.gz", "-C", ".")
++            core_gem_path = "#{DOWNLOADS_DIR}/core_gems"
++            core_gems_files = ["#{core_gem_path}/00-bundler-2.1.4.gem",
```

Comment:
Why do we need these binaries?  Can we not build them ourselves?  What does Fedora or others do for this?